lib/ruby/truffle/truffle/bigdecimal.rb

```diff
@@ -181,6 +189,70 @@ def frac
       self
     end
   end
+
+  def to_s(format = 'E')
+    if finite?
+      spaces       = format.to_i
+      prefix       = case
+                     when self > 0
+                       if [' ', '+'].include? format[0]
+                         format[0]
+                       else
+                         ''
+                       end
+                     when self < 0
+                       '-'
+                     when self == 0
+                       ''
+                     end
+      float_format = format[-1] == 'F'
+
+      unscaled_value = unscaled
+      exponent_value = self.exponent
+
+      if float_format
+        before_dot, after_dot = case
+                                when exponent_value > unscaled_value.size
+                                  [unscaled_value + '0' * (exponent_value - unscaled_value.size),
+                                   '0']
+                                when exponent_value <= 0
+                                  ['0',
+                                   '0' * exponent_value.abs + unscaled_value]
+                                else
+                                  after_dot = unscaled_value[exponent_value..-1]
+
+                                  [unscaled_value[0...exponent_value],
+                                   after_dot.empty? ? '0' : after_dot]
+                                end
+
+        prefix << add_spaces_to_s(before_dot, true, spaces) << '.' << add_spaces_to_s(after_dot, false, spaces)
```

Comment:
`Kernel#format` ? I think the current version is better in that case (`format` is not idiomatic IMO).
Interpolation is a bit hard to read because there is no space between tokens, but it shows more clearly the order.